src/components/CountrySelector/CountrySelector.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useEffect, useState } from "react";
+import { i18n, TextField, useTheme } from "@apisuite/fe-base";
+import countries from "i18n-iso-countries";
+import { Autocomplete } from "@material-ui/lab";
+
+import { CountrySelectorProps } from "./types";
+import useStyles from "./styles";
+
+const CountrySelector: React.FC<CountrySelectorProps> = ({
+  countrySelectionHandler,
+  selectedCountry,
+}) => {
+  const classes = useStyles();
+  const { spacing } = useTheme();
+
+  /* Country selector set-up */
+
+  const userLanguageCode = i18n.language.substr(0, 2);
+
+  // Retrieves all countries in the user's currently selected language (e.g., 'Brasil' in pt-PT, 'Brazil' in en-UK/US)
+  countries.registerLocale(require(`i18n-iso-countries/langs/${userLanguageCode}.json`));
+
+  const countryCodesAndNames = countries.getNames(userLanguageCode);
+  const autoCompleteOptions = Object.entries(countryCodesAndNames).map((country) => ({
+    countryCode: country[0],
+    countryName: country[1],
+  }));
```

Comment:
Not critical but this can be achieved with a single `for..in` loop instead of using 2 (.entries and .map)